### PR TITLE
Fix nextTurn for when players have no units (only happens in 2v2)

### LINKED ISSUE #273: Modify the change of turn in the server.

### PR DESCRIPTION
**Pull request by hilalcomak**
Change Game nextTurn to skip players with no units.

Closes #273

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs22/game/Game.java [hunk 1]
```diff
@@ -23,6 +23,7 @@ public class Game {
     private final GameMode gameMode;
     private final GameType gameType;
     private final Map<String, PlayerDecorator> decoratedPlayers;
+    private int turnIndex;
```
Comment (apple00juice) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/game/Game.java:26]: Maybe we should encapsulate turnIndex and turnNumber into a class where you set the total number of players in the constructor.
With two getters for turnIndex and turnNumber.
And one method for nextTurn.

I am just a bit scared that we miss to increase one field somewhere and then it doesn't work anymore.. :|

**Review by apple00juice — APPROVED** [review]
Rest looks super! 😄

File: src/main/java/ch/uzh/ifi/hase/soprafs22/game/Game.java (hunk 1 above)
Comment (hilalcomak) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/game/Game.java:27]: Done, let me know your opinion.

Comment (apple00juice) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/game/Game.java:27]: Perfect ty! I think that's saver 👍

**Review by apple00juice — APPROVED** [review]
👍

**Review by paolovic — APPROVED** [review]

**Pull request merged**